R/comparePlot.R

```diff
@@ -141,133 +142,111 @@ comparePlot <- function(compareDF,
 
     if (all(c("xp","yp") %in% colnames(compareDF))) {
         sigMeasurePlot <- TRUE
-        compareDF$group <- NA
-        xp_boolean_vec <- subset(compareDF, select = xp, drop = TRUE) <= pThreshold
-        yp_boolean_vec <- subset(compareDF, select = yp, drop = TRUE) <= pThreshold
-        compareDF$group <- ifelse(xp_boolean_vec,
-                                  ifelse(yp_boolean_vec, "Common", "X Unique"),
-                                  ifelse(yp_boolean_vec, "Y Unique", "Not Significant"))
-
-        # Set an order field to control order of plotting for ggplot
-        # Plot order is 1) Not Significant plotted first, 2) randomly plot X and Y Unique
-        # then plot Common last
-        compareDF$order <- sample.int(nrow(compareDF)) + # Add random order (seeded for reproducibility)
-            nrow(compareDF) * (compareDF$group == "Common") +  # Assign common a high value to sort last
-            -1*nrow(compareDF) * (compareDF$group == "Not Significant") # Assign NotSig group neg values to sort first
-        compareDF <- compareDF %>% dplyr::arrange(order)
+        compareDF <- compareDF %>%
+            dplyr::mutate(group = ifelse(xp <= pThreshold,
+                                         ifelse(yp <= pThreshold, "Common", "X Unique"),
+                                         ifelse(yp <= pThreshold, "Y Unique", "Not Significant")),
+                          group = factor(group,
+                                         levels = c("Common", "X Unique", "Y Unique", "Not Significant")))
     }
 
-    # plotType
-    if (plotType == "canvasXpress") {
+    y_range <- compareDF %>%
+        dplyr::select(ylabel) %>%
+        dplyr::summarise(across(everything(), list(min, max))) %>%
+        dplyr::rename_with(~ c("min", "max"))
+    if (plotType == "canvasxpress") {
         # adding alpha to colors
         symbolFill <- sapply(symbolFill, rgbaConversion, alpha = alpha, USE.NAMES = FALSE)
```

Comment:
alpha needs to be renamed to transparency. Connie had mentioned in one of the comments for plotDispersion

Reply:
done